packages/x-data-grid/src/components/GridRow.tsx

```diff
@@ -357,6 +357,11 @@ const GridRow = React.forwardRef<HTMLDivElement, GridRowProps>(function GridRow(
       indexRelativeToAllColumns,
     );
 
+    if (!column) {
+      // See https://github.com/mui/mui-x/issues/13719
+      return null;
+    }
```

Comment:
This feels like it hides something. 

```suggestion
```

Looking a bit at it, it seems that the issue is that the frozen render context isn't reset when the column changes.

```diff
diff --git a/packages/x-data-grid/src/hooks/features/virtualization/useGridVirtualScroller.tsx b/packages/x-data-grid/src/hooks/features/virtualization/useGridVirtualScroller.tsx
index 334d65759..d87187caa 100644
--- a/packages/x-data-grid/src/hooks/features/virtualization/useGridVirtualScroller.tsx
+++ b/packages/x-data-grid/src/hooks/features/virtualization/useGridVirtualScroller.tsx
@@ -277,6 +277,7 @@ export const useGridVirtualScroller = () => {
   const forceUpdateRenderContext = () => {
     const inputs = inputsSelector(apiRef, rootProps, enabled, enabledForColumns);
     const nextRenderContext = computeRenderContext(inputs, scrollPosition.current, scrollCache);
+    frozenContext.current = undefined;
     updateRenderContext(nextRenderContext);
   };
```

Reply:
Nice catch!